common/bounded-executor/src/lib.rs

```diff
@@ -8,20 +8,24 @@
 //! `capacity`.
 
 use futures::future::{Future, FutureExt};
-use futures_semaphore::{Permit, Semaphore};
-use tokio::{runtime::Handle, task::JoinHandle};
+use std::sync::Arc;
+use tokio::{
+    runtime::Handle,
+    sync::{OwnedSemaphorePermit, Semaphore},
+    task::JoinHandle,
+};
 
 #[derive(Clone, Debug)]
 pub struct BoundedExecutor {
-    semaphore: Semaphore,
+    semaphore: Arc<Semaphore>,
     executor: Handle,
 }
 
 impl BoundedExecutor {
     /// Create a new `BoundedExecutor` from an existing tokio [`Handle`]
     /// with a maximum concurrent task capacity of `capacity`.
     pub fn new(capacity: usize, executor: Handle) -> Self {
-        let semaphore = Semaphore::new(capacity);
+        let semaphore = Arc::new(Semaphore::new(capacity));
```

Comment:
Is this remove `futures-semaphore` something found in the fuzzing, or just for simplification?

Reply:
just some refactoring after stream-ratelimiter no longer could use it and it turned out to be unnecessary now in bounded-executor as well